rust/plugin/src/diff/scene_differ.rs

```diff
@@ -0,0 +1,583 @@
+				// Instance properties are always set, regardless of the default value, so this is always empty
+				_ => "".to_string(),
+			},
+            // Otherwise, get the value from the property.
+            |val| val.get_value(),
+        );
+
+        Some(Self::get_varstr_value(val))
+    }
+
+    /// Returns a [PropertyDiff] comparing the old property value versus the new one.
+    /// Returns [None] if neither node is valid, or if the value has not meaningfully changed.
+    fn get_property_diff(
+        &self,
+        prop: &String,
+        old_node: Option<&GodotNode>,
+        new_node: Option<&GodotNode>,
+        old_scene: Option<&GodotScene>,
+        new_scene: Option<&GodotScene>,
+    ) -> Option<PropertyDiff> {
+        // If neither node is valid, there's no valid property diff.
+        if new_node.is_none() && old_node.is_none() {
+            return None;
+        };
+
+        // Slightly weird hack: Diff against the default instead of the normal property.
+        let old_value = Self::get_varstr_or_default(prop, old_node);
+        let new_value = Self::get_varstr_or_default(prop, new_node);
+
+        // Skip in case of no changes
+        if !self.did_prop_change(old_value.as_ref(), new_value.as_ref(), old_scene, new_scene) {
+            return None;
+        }
+
+        // Expensive: Load any ext resources and turn them into Variants
+        let old = old_value
+            .as_ref()
+            .map(|v| self.get_prop_value(&v, old_scene, true, prop == "script"));
+        let new = new_value
+            .as_ref()
+            .map(|v| self.get_prop_value(&v, new_scene, false, prop == "script"));
+
+        return Some(PropertyDiff::new(
+            prop.clone(),
+            // We check for node add or remove intentionally, here, because otherwise we're just diffing a Modified prop against
+            // the default value retrieved earlier.
+            match (old_node, new_node) {
+                (None, Some(_)) => ChangeType::Added,
+                (Some(_), None) => ChangeType::Removed,
+                (_, _) => ChangeType::Modified,
+            },
+            old,
+            new,
+        ));
+    }
+
+    /// Check deeply to see if a subresource has changed.
+    fn did_sub_resource_change(
+        &self,
+        old_resource: Option<&SubResourceNode>,
+        new_resource: Option<&SubResourceNode>,
+        old_scene: &GodotScene,
+        new_scene: &GodotScene,
+    ) -> bool {
+        let (old_resource, new_resource) = match (old_resource, new_resource) {
+            (None, None) => return false,         // subresource never existed
+            (None, Some(_)) => return true,       // subresource added
+            (Some(_), None) => return true,       // subresource removed
+            (Some(old), Some(new)) => (old, new), // keep looking
+        };
+
+        // If ID or type has changed, subresource has definitely changed.
+        if old_resource.id != new_resource.id
```

Comment:
we shouldn't do the id check here, as Godot loves to randomize the IDs on both external and internal resources when resaving a scene or a resource.